pkg/targets/adapter/googlecloudpubsubtarget/adapter.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package googlecloudpubsubtarget
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
+
+	"cloud.google.com/go/pubsub"
+	"google.golang.org/api/option"
+
+	"github.com/triggermesh/triggermesh/pkg/metrics"
+	targetce "github.com/triggermesh/triggermesh/pkg/targets/adapter/cloudevents"
+)
+
+// NewTarget adapter implementation
+func NewTarget(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
+	env := envAcc.(*envAccessor)
+	logger := logging.FromContext(ctx)
+	replier, err := targetce.New(env.Component, logger.Named("replier"),
+		targetce.ReplierWithStatefulHeaders(env.BridgeIdentifier),
+		targetce.ReplierWithStaticResponseType("io.triggermesh.googlecloudpubsubtarget.response"),
+		targetce.ReplierWithPayloadPolicy(targetce.PayloadPolicy(env.CloudEventPayloadPolicy)))
+	if err != nil {
+		logger.Panicf("Error creating CloudEvents replier: %v", err)
+	}
+
+	mt := &pkgadapter.MetricTag{
+		ResourceGroup: "googlecloudpubsubtargets",
+		Namespace:     envAcc.GetNamespace(),
+		Name:          envAcc.GetName(),
+	}
+
+	psCli, err := pubsub.NewClient(ctx, env.TopicName.Project,
+		option.WithCredentialsJSON(env.ServiceAccountKey),
+	)
+	if err != nil {
+		logger.Panicw("Failed to create Google Cloud Pub/Sub API client", zap.Error(err))
+	}
+
+	t := psCli.Topic(env.TopicName.Resource)
+	return &googlecloudpubsubtargetAdapter{
+		topic: t,
+
+		replier:          replier,
+		ceClient:         ceClient,
+		logger:           logger,
+		mt:               mt,
+		sr:               metrics.MustNewEventProcessingStatsReporter(mt),
+		discardCEContext: env.DiscardCEContext,
+	}
+}
+
+var _ pkgadapter.Adapter = (*googlecloudpubsubtargetAdapter)(nil)
+
+type googlecloudpubsubtargetAdapter struct {
```

Comment:
How about renaming this structure to idiomatic `googleCloudPubSubTargetAdapter` or, for simplicity, making it `adapter`?